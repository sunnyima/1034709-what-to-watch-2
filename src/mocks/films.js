const films = [
  {
    id: 1,
    name: `Побег из Шоушенка`,
    posterImage: `https://www.kino-teatr.ru/movie/posters/big/8/25308.jpg`,
    previewImage: `http://t3.gstatic.com/images?q=tbn:ANd9GcTY9gSsZBmGbhNZrkkFgmT1npIV_px6Wuc09eTaUFEpoyVJAOb0`,
    backgroundImage: `https://www.kino-teatr.ru/movie/poster/25308/37333.jpg`,
    backgroundColor: `#ffffff`,
    videoLink: `https://www.youtube.com/watch?v=1R2EQ__1w8Q`,
    previewVideoLink: `https://www.youtube.com/watch?v=1R2EQ__1w8Q`,
    description: `Бухгалтер Энди Дюфрейн обвинён в убийстве собственной жены и её любовника. Оказавшись в тюрьме под 
                  названием Шоушенк, он сталкивается с жестокостью и беззаконием, царящими по обе стороны решётки. 
                  Каждый, кто попадает в эти стены, становится их рабом до конца жизни. Но Энди, обладающий живым умом и 
                  доброй душой, находит подход как к заключённым, так и к охранникам, добиваясь их особого к себе расположения.`,
    rating: 9.1,
    director: `Фрэнк Дарабонт`,
    starring: [`Тим Роббинс`, `Морган Фриман`, `Боб Гантон`, `Уильям Сэдлер`, `Клэнси Браун`, `Гил Беллоуз`],
    runTime: 142,
    genre: `Драма`,
    released: 1994,
    isFavorite: false,
  },
  {
    id: 2,
    name: `Зелёная миля`,
    posterImage: `https://www.kino-teatr.ru/movie/poster/26086/46544.jpg`,
    previewImage: `https://www.kino-teatr.ru/movie/posters/big/6/26086.jpg`,
    backgroundImage: `https://www.kino-teatr.ru/movie/poster/26086/46545.jpg`,
    backgroundColor: `#ffffff`,
    videoLink: `https://www.youtube.com/watch?v=0MjG7uZJj4g`,
    previewVideoLink: `https://www.youtube.com/watch?v=0MjG7uZJj4g`,
    description: `Обвиненный в страшном преступлении, Джон Коффи оказывается в блоке смертников тюрьмы «Холодная гора». 
                  Вновь прибывший обладал поразительным ростом и был пугающе спокоен, что, впрочем, никак не влияло на 
                  отношение к нему начальника блока Пола Эджкомба, привыкшего исполнять приговор.`,
    rating: 9.1,
    director: `Фрэнк Дэрабонт`,
    starring: [`Том Хэнкс`, `Майкл Кларк Дункан`, `Дэвид Морс`],
    runTime: 189,
    genre: `Драма`,
    released: 1999,
    isFavorite: false,
  },
  {
    id: 3,
    name: `Список Шиндлера`,
    posterImage: `https://www.kino-teatr.ru/kino/movie/hollywood/19751/foto/`,
    previewImage: `https://www.kino-teatr.ru/movie/poster/19751.jpg`,
    backgroundImage: `https://www.kino-teatr.ru/movie/kadr/19751/339205.jpg`,
    backgroundColor: `#ffffff`,
    videoLink: `https://www.youtube.com/watch?v=Li5pm1dcSN0`,
    previewVideoLink: `https://www.youtube.com/watch?v=Li5pm1dcSN0`,
    description: `Лента рассказывает реальную историю загадочного Оскара Шиндлера, члена нацистской партии, преуспевающего 
                  фабриканта, спасшего во время Второй мировой войны более тысячи ста евреев. Это триумф одного человека, 
                  не похожего на других, и драма тех, кто, благодаря ему, выжил в ужасный период человеческой истории.`,
    rating: 9.0,
    director: `Стивен Спилберг`,
    starring: [`Лиам Нисон`, `Бен Кингсли`, `Рэйф Файнс`],
    runTime: 197,
    genre: `Драма`,
    released: 1993,
    isFavorite: false,
  },
  {
    id: 4,
    name: `Унесённые ветром`,
    posterImage: `https://www.kino-teatr.ru/movie/poster/24253/30269.jpg`,
    previewImage: `https://www.kino-teatr.ru/movie/posters/big/3/24253.jpg`,
    backgroundImage: `https://www.kino-teatr.ru/news/10787/105183.jpg`,
    backgroundColor: `#ffffff`,
    videoLink: `https://www.youtube.com/watch?v=KfPCpgEVVsI`,
    previewVideoLink: `https://www.youtube.com/watch?v=KfPCpgEVVsI`,
    description: `Незабываемый любовный роман времен американской Гражданской войны. Бессмертным героям этой эпической 
                  саги Скарлетт, Ретт, Эшли, Мелани, Матушка и Присси — удалось покорить не одно поколение кинозрителей.`,
    rating: 9.4,
    director: `Виктор Флеминг`,
    starring: [`Вивьен Ли`, `Кларк Гейбл`, `Лесли Хоуард`, `Оливия де Хэвилленд`],
    runTime: 238,
    genre: `Мелодрама`,
    released: 1939,
    isFavorite: false,
  },
  {
    id: 5,
    name: `Форрест Гамп`,
    posterImage: `https://www.kino-teatr.ru/movie/poster/22041/75761.jpg`,
    previewImage: `https://www.kino-teatr.ru/movie/posters/big/1/22041.jpg`,
    backgroundImage: `https://www.kino-teatr.ru/movie/poster/22041/40138.jpg`,
    backgroundColor: `#ffffff`,
    videoLink: `https://www.youtube.com/watch?v=gB7wy6YGCGA`,
    previewVideoLink: `https://www.youtube.com/watch?v=gB7wy6YGCGA`,
    description: `От лица главного героя Форреста Гампа, слабоумного безобидного человека с благородным и открытым сердцем, 
                  рассказывается история его необыкновенной жизни. Фантастическим образом превращается он в известного футболиста, 
                  героя войны, преуспевающего бизнесмена. Он становится миллиардером, но остается таким же бесхитростным, 
                  глупым и добрым. Форреста ждет постоянный успех во всем, а он любит девочку, с которой дружил в детстве, 
                  но взаимность приходит слишком поздно.`,
    rating: 9.2,
    director: `Роберт Земекис`,
    starring: [`Том Хэнкс`, `Робин Райт`, `Гэри Синиз`, `Микелти Уильямсон`],
    runTime: 202,
    genre: `Мелодрама`,
    released: 1994,
    isFavorite: false,
  }
];

export default films;
