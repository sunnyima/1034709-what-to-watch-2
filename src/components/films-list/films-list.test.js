import React from 'react';
import renderer from 'react-test-renderer';
import FilmsList from "./films-list";
import films from "../../mocks/films";

it(`FilmList correctly renders after relaunch`, () => {
  const tree = renderer.create(<FilmsList
    films={films}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});

