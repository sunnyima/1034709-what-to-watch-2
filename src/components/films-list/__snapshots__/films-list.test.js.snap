// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmList correctly renders after relaunch 1`] = `
Array [
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Побег из Шоушенка"
        height="175"
        srcSet="http://t3.gstatic.com/images?q=tbn:ANd9GcTY9gSsZBmGbhNZrkkFgmT1npIV_px6Wuc09eTaUFEpoyVJAOb0"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href=""
      >
        Побег из Шоушенка
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Зелёная миля"
        height="175"
        srcSet="https://www.kino-teatr.ru/movie/posters/big/6/26086.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href=""
      >
        Зелёная миля
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Список Шиндлера"
        height="175"
        srcSet="https://www.kino-teatr.ru/movie/poster/19751.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href=""
      >
        Список Шиндлера
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Унесённые ветром"
        height="175"
        srcSet="https://www.kino-teatr.ru/movie/posters/big/3/24253.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href=""
      >
        Унесённые ветром
      </a>
    </h3>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <img
        alt="Форрест Гамп"
        height="175"
        srcSet="https://www.kino-teatr.ru/movie/posters/big/1/22041.jpg"
        width="280"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href=""
      >
        Форрест Гамп
      </a>
    </h3>
  </article>,
]
`;
